server.js

```diff
@@ -307,3 +315,20 @@ Server.listening = function () {
 
     Server.ready = 1;
 };
+
+Server.close = function () {
+    var list = Server.smtp_server_listeners;
+    for (var i=0; i < list.length; i++) {
+        list[i].server.close();
+    }
+
+    if (!Server.cluster) {
+        out.stop_queue();
+        return;
+    }
+
+    for (var id in cluster.workers) {
+        cluster.workers[id].send({event: 'outbound.stop_queue'});
```

Comment:
In order for this to work you need to add it here too:

https://github.com/roosit/Haraka/blob/smtpeshka/outbound.js#L128